### PR TITLE
ImportDashboard: Fixes issue with importing dashboard and name ending up in uid

### LINKED ISSUE #43097: 8.3.1: Imported externally shared dashboards gets name instead of ID under datasource uid
Imported dashboards that was exported with the sharing externally option enabled gets name of datasource instead of uid of datasource saved under all datasource settings, e.g. option "uid" under "datasource" contains string instead of unique ID. The incorrect value saved under datasource - uid strangely enough mostly works, it is not until after trying to add an alert to one of the graphs that Grafana cannot save the dashboard because it cannot find the datasource in question, msg "Failed to save dashboard" & error "data source not found".

**What happened**:
Imported dashboard gets datasource blocks like this:
```
        "datasource": {
          "type": "cloudwatch",
          "uid": "CloudWatch - Test"
        },
```

**What you expected to happen**:
Imported dashboard should get datasource blocks like this:
```
      "datasource": {
        "type": "cloudwatch",
        "uid": "d8dcG1cGz"
      },
```

**How to reproduce it (as minimally and precisely as possible)**:
Export with sharing externally enabled and then import the dashboard under Grafana 8.3.1.

**Anything else we need to know?**:
It looks like graphs works even when a name is used instead of a UID but when you try to add an alert things blows up with "Failed to save dashboard". The reason for which is that the datasource cannot be found because of the invalid uid value saved by Grafana.

```
t=2021-12-14T10:16:51+0000 lvl=eror msg="Failed to save dashboard" logger=context userId=82 orgId=1 uname=X error="data source not found" remote_addr=172.0.0.1
```

Note that we are not using Grafana 8 Alert yet.

**Environment**:
- Grafana version: 8.3.1
- Data source type & version: CloudWatch, Elasticsearch & InfluxDB
- OS Grafana is installed on: Amazon Linux 2

### PR DESCRIPTION
**Pull request by torkelo**
Fixes #43097

Needs a unit test, but I have to break for the day. Will add later if I can find a spare moment.

### CODE REVIEW

File: public/app/features/manage-dashboards/components/ImportDashboardForm.tsx [hunk 1]
```diff
@@ -124,7 +124,7 @@ export const ImportDashboardForm: FC<Props> = ({
                     noDefault={true}
                     placeholder={input.info}
                     pluginId={input.pluginId}
-                    current={current[index]?.name}
+                    current={current[index]?.uid}
```
Comment (hugohaggmark) [inline public/app/features/manage-dashboards/components/ImportDashboardForm.tsx:127]: Will this work for exports from older Grafana versions?

**Review by hugohaggmark — APPROVED** [review]
Looks great, some tests would be great as you already mentioned 😄

File: public/app/features/manage-dashboards/components/ImportDashboardForm.tsx (hunk 1 above)
Comment (axelavargas) [reply public/app/features/manage-dashboards/components/ImportDashboardForm.tsx:127]: It seems to work, I tested importing a dashboard from 7.5.5 with Prometheus Datasource(no `datasource.uid `there) and it worked, I think the backend is taking into account when we don't provision a datasource.uid

**Review by axelavargas — APPROVED** [review]
👏🏾 👏🏾 great job @torkelo, I gave it a spin, and worked as expected, I also did the test of importing dashboards from older versions and it did not generate any error 🚀

**Pull request merged**